src/app.js

```diff
@@ -1 +1,47 @@
 'use strict';
+
+const { TOTAL_DIGITS } = require('./constants/constants');
+const generateNumbers = require('./generateNumbers');
+const {
+  terminal,
+  getInput,
+  printOut,
+} = require('./ask');
+const calculateBullsAndCows = require('./calculateBullAndCows');
+
+const bullsAndCowsGame = async() => {
+  const randomNumber = generateNumbers();
+  let guessNumber = '';
+  let attempts = 0;
+
+  while (guessNumber !== randomNumber) {
+    guessNumber = await getInput('Enter your guess: ');
+
+    const isValidInput = new Set(guessNumber).size !== TOTAL_DIGITS;
+
+    if (isValidInput) {
+      printOut(`Enter ${TOTAL_DIGITS}-digits number with unique digits.`);
+      continue;
+    }
+
+    attempts += 1;
```

Comment:
I would check the users message if it acually doesn't have nothing just numbers as well, but it is not necessary according to the task.
I completely don't know how this is works `new Set(guessNumber).size !== TOTAL_DIGITS` 😂 `.size` for sets is the same as `.length` for arrays, you compare one digit number with total digits. I'am not sure but i suggest that it throws this message on every step. Does your code work? 🧐
 Also if the values isn't unique, the attempts not gonna be incremented as you quit this iteration of while inside of the block with `continue.`
 :)